lib/plugins/bayesian-optimization/presentation/dialogs/bayesian_optimization_training_dialog_bloc.dart

```diff
@@ -228,34 +226,30 @@ class BOTrainingDialogBloc
     on<DesiredBooleanValueUpdated>(_onDesiredBooleanValueUpdated);
   }
 
-  void _onDatasetSelected(
-      DatasetSelected event, Emitter<BOTrainingDialogState> emit) {
+  void _onDatasetSelected(DatasetSelected event, Emitter<BOTrainingDialogState> emit) {
     var availableFeatures = [''];
     if (event.dataset.toString() == 'Protein') {
```

Comment:
If you take a look at the `BiocentralEntityTypeSelection` you'll see that it actually already returns a type, so you don't need to use strings for type checking and can even omit the whole explicit type checking (as, in theory, the plugin should work with all data types in the future)